tests/rptest/tests/cluster_metrics_test.py

```diff
@@ -90,42 +90,64 @@ def _failover(self) -> ClusterNode:
             hosts=started_hosts,
             check=lambda node_id: node_id != self.redpanda.idx(prev))
 
-    def _get_value_from_samples(self, samples: MetricSamples):
-        """
-        Extract the metric value from the samples.
-        Only one sample is expected as cluster level metrics have no labels.
-        """
-        assert len(samples.samples) == 1
-        return samples.samples[0].value
-
-    def _assert_cluster_metrics(
+    def _get_metrics_value_from_node(self, node: ClusterNode, pattern: str):
+        samples = self._get_metrics_from_node(node, [pattern])
+        assert pattern in samples
+        value = samples[pattern].samples[0].value
+        self.logger.info(f"Found metric value {value} for {pattern}")
+        return value
+
+    def _wait_until_metric_has_value(self, node: ClusterNode, pattern: str,
+                                     value):
+        return wait_until(
```

Comment:
nit: drop the `return`? Reads like a typo at first since this could be `wait_until_result()`